### PR TITLE
Feature/user admin

### PR DESCRIPTION
**Pull request by minahil3955**
Working on Admin Panel
1.  Created active_admin
2.  Add New User functionality Implemented

### CODE REVIEW

File: app/admin/dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    #     end
+    #   end
+
+    #   column do
+    #     panel "Info" do
+    #       para "Welcome to ActiveAdmin."
+    #     end
+    #   end
+    # end
+  end # content
```
Comment (muhammad-ahmad-shafiq) [inline app/admin/dashboard.rb]: Remove this commented code.

File: .byebug_history [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@resource.roles.count
+@resource.roles
+@resource
+@resource.roles.first.name
+@resource.roles.first
+@resource.roles
+@resource
+n
+r.name
+n
```
Comment (muhammad-ahmad-shafiq) [inline .byebug_history]: Why is this file tracked by git?

File: app/models/admin_user.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AdminUser < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, 
+         :recoverable, :rememberable, :validatable
+end
```
Comment (muhammad-ahmad-shafiq) [inline app/models/admin_user.rb]: Why do we need to add adminuser?

File: app/models/phase.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Phase < ApplicationRecord
   validates :due_date, :status, presence: true
-  validate due_date_cannot_be_in_the_past
+  validate :due_date_cannot_be_in_the_past
```
Comment (muhammad-ahmad-shafiq) [inline app/models/phase.rb]: Wrong name

File: config/initializers/setup_mail.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+ActionMailer::Base.smtp_settings = {
+    :address              => "smtp.gmail.com",
+    :port                 => 587,
+    :domain               => "mail.google.com",
+    :user_name            => "minahil.mustafa@devsinc.com",
+    :password             => "minahil3955",
+    :authentication       => "plain",
+    :enable_starttls_auto => true
+  }
+
```
Comment (muhammad-ahmad-shafiq) [inline config/initializers/setup_mail.rb:10]: Do you know about environment variables?

File: db/migrate/20210409091635_create_active_admin_comments.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      t.references :resource, polymorphic: true
+      t.references :author, polymorphic: true
+      t.timestamps
+    end
+    add_index :active_admin_comments, [:namespace]
+  end
+
+  def self.down
+    drop_table :active_admin_comments
+  end
```
Comment (muhammad-ahmad-shafiq) [inline db/migrate/20210409091635_create_active_admin_comments.rb:15]: why this type of name of a table?

**Review by muhammad-ahmad-shafiq — CHANGES_REQUESTED** [review]

File: .byebug_history (hunk 2 above)
Comment (minahil3955) [reply .byebug_history]: Added this file in .gitignore

File: app/models/phase.rb (hunk 4 above)
Comment (minahil3955) [reply app/models/phase.rb]: Name Changed to due_date_in_future

File: db/migrate/20210409091635_create_active_admin_comments.rb (hunk 6 above)
Comment (minahil3955) [reply db/migrate/20210409091635_create_active_admin_comments.rb:15]: it was built in when, generated Active Admin. If I will change it,  will face issues where ever it is used.

File: config/initializers/setup_mail.rb (hunk 5 above)
Comment (minahil3955) [reply config/initializers/setup_mail.rb:10]: Env variable describes our environment or to do some configurations on specific environment.

File: app/models/admin_user.rb (hunk 3 above)
Comment (minahil3955) [reply app/models/admin_user.rb]: It was built in generated model

File: config/initializers/setup_mail.rb (hunk 5 above)
Comment (minahil3955) [reply config/initializers/setup_mail.rb:10]: This configuration is not needed for the time being but may be in future.

**Review by muhammad-ahmad-shafiq — APPROVED** [review]

**Pull request merged**